### ISSUE TITLE
pip_setup PRs being opened and then autoclosed

### ISSUE THREAD

**Issue by djmitche** [renovatebot/renovate#4760]
See
* https://github.com/taskcluster/tc-admin/pull/38
* https://github.com/taskcluster/tc-admin/pull/39
* https://github.com/taskcluster/tc-admin/pull/40
* https://github.com/taskcluster/tc-admin/pull/41
* https://github.com/taskcluster/tc-admin/pull/42

### Which Renovate are you using? CLI, App, or Pro

App

### Which platform are you using? GitHub, GitLab, Bitbucket Azure DevOps

GitHub

### Have you checked the logs? Don't forget to include them if relevant

https://app.renovatebot.com/dashboard#github/taskcluster/tc-admin/106993579

`WARN: Failed to read setup.py file(packageFile="setup.py")` due to what looks like a docker failure?  The error message says success so I'm not sure what to make of it.

### What is your question?

More of a bug report: if docker fails in trying to run setup.py, I would think the run should fail, vs. deciding there are no updates to do and autoclosing PRs. But perhaps I have something wrong in the config for this repo?

**Comment by rarkins** [comment]
Thanks for the help diagnosing this. This was happening because:
- pip_setup extraction is set to 5s timeout because it should take milliseconds and we don't want runaway python processes, but
- when the renovate/pip image is not present, it needs to be pulled by the Docker daemon and that took longer than 5s
- the extraction process was killed, but we don't abort failed pip_setup extractions because there are too many repositories for which this might mean "permanent" aborts

So when the renovate/pip image was present on the Renovate worker then it succeeded and created a PR, but when it was not then we got ignored timeouts and then autoclosed PRs. I have resolved this by adding a pre-step that pulls the image first, without any timeout.
